dataworkspace/dataworkspace/apps/applications/utils.py

```diff
@@ -907,3 +913,149 @@ def sync_quicksight_permissions(
+    header = Sender(
+        {'id': hawk_id, 'key': hawk_key, 'algorithm': 'sha256'},
+        url,
+        method,
+        content=body,
+        content_type=content_type,
+    ).request_header
+
+    response = requests.request(
+        method,
+        url,
+        data=body,
+        headers={'Authorization': header, 'Content-Type': content_type},
+    )
+    return response.status_code, response.content
+
+
+def publish_to_iam_role_creation_channel(user):
+    pass
+
+
+@celery_app.task()
+def sync_activity_stream_sso_users():
+    try:
+        with cache.lock(
+            "activity_stream_sync_last_published", blocking_timeout=0, timeout=1800
+        ):
+            _do_sync_activity_stream_sso_users()
+    except redis.exceptions.LockError:
+        logger.info(
+            "activity_stream_sync_last_published: Unable to grab lock - running on another instance?"
+        )
+
+
+def _do_sync_activity_stream_sso_users():
+    last_published = cache.get(
+        'activity_stream_sync_last_published', datetime.datetime.utcfromtimestamp(0)
+    )
+
+    endpoint = f'{settings.ACTIVITY_STREAM_BASE_URL}/v3/activities/_search'
+    ten_seconds_before_last_published = last_published - datetime.timedelta(seconds=10)
+
+    query = {
+        "query": {
+            "bool": {
+                "filter": [
+                    {"term": {"object.type": "dit:StaffSSO:User"}},
+                    {
+                        "range": {
+                            "published": {
+                                "gte": f"{ten_seconds_before_last_published.strftime('%Y-%m-%dT%H:%M:%S')}"
+                            }
+                        }
+                    },
+                ]
+            }
+        },
+        "sort": [{"published": "asc"}],
+    }
+
+    while True:
+        try:
+            status_code, response = hawk_request('GET', endpoint, json.dumps(query),)
+        except HawkException as e:
+            logger.error('Failed to call activity stream with error %s', e)
+            break
+
+        if status_code != 200:
+            raise Exception(f'Failed to fetch SSO users: {response}')
+
+        response_json = json.loads(response)
+
+        if 'failures' in response_json['_shards']:
+            raise Exception(
+                f"Failed to fetch SSO users: {','.join(response_json['_shards']['failures'])}"
+            )
+
+        records = response_json['hits']['hits']
+
+        if not records:
+            break
+
+        logger.info('Fetched %d from activity stream', len(records))
+
+        for record in records:
+            obj = record['_source']['object']
+
+            user_id = obj['dit:StaffSSO:User:userId']
+            email = obj['dit:emailAddress'][0]
+
+            User = get_user_model()
+            try:
+                user = User.objects.get(username=email)
+            except User.DoesNotExist:
+                logger.info('User with SSO id %s does not exist, creating...', user_id)
```

Comment:
This isn't quite accurate ... 
```suggestion
                logger.info('User with email address %s does not exist, creating...', email)
```